src/Features/ERDDAP/ForecastsMetadata/Menu/index.tsx

```diff
@@ -1,38 +1,62 @@
+/**
+ * Dropdown menu for accessing forecasts via buoy barn
+ */
 import * as React from "react"
-import { connect } from "react-redux"
 import { Link } from "react-router-dom"
-import { Dropdown, DropdownMenu, DropdownToggle } from "reactstrap"
+import { Dropdown, DropdownMenu, DropdownToggle, NavItem } from "reactstrap"
 
 import { paths } from "Shared/constants"
-import { StoreState } from "Shared/constants/store"
 import { urlPartReplacer } from "Shared/urlParams"
 
+import { useForecasts } from "../../hooks"
 import { ForecastSource } from "../../types"
 
-const initialState = {
-  dropdownOpen: false
+export interface Props {
+  platformId: string
 }
 
-type State = Readonly<typeof initialState>
+/**
+ * Wraps and loads the forecast menu
+ */
+export const ForecastDropdown: React.FunctionComponent<Props> = ({ platformId }) => {
+  const { isLoading, data } = useForecasts()
 
-export interface Props {
-  platformId: string
+  if (isLoading) {
+    return (
+      <NavItem>
+        <div className="nav-link">Forecasts loading</div>
+      </NavItem>
+    )
+  }
+
+  if (data) {
+    return <ForecastDropdownBase platformId={platformId} forecasts={data} />
```

Comment:
Codacy found an issue: ['ForecastDropdownBase' was used before it was defined.](https://app.codacy.com/manual/gmri/Neracoos-1-Buoy-App/pullRequest?prid=6166252)